fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -1384,4 +1386,242 @@ private SqlQueryData processMissingParm(Class<?> resourceType, QueryParameter qu
+                                        .append(previousParm.getModifierResourceTypeName() + "_TOKEN_VALUES_V AS " + chainedParmVar + WHERE);
+                    }
+
+                    // Build this piece: CPx.PARAMETER_NAME_ID = <code-id> AND CPx.STR_VALUE IN
+                    appendMidChainParm(selectSegments, currentParm, chainedParmVar);
+
+                    // Build this piece: (SELECT 'resource-type-name' || '/' || CLR<x+1>.LOGICAL_ID ...
+                    selectSegments.append(LEFT_PAREN);
+                    appendInnerSelect(selectSegments, currentParm, currentParm.getModifierResourceTypeName(),
+                        nextChainedResourceVar, nextChainedLogicalResourceVar, nextChainedParmVar);
+                    whereClauseSegment.append(RIGHT_PAREN);
+
+                    if (!chainedParmProcessed) {
+                        chainedParmProcessed = true;
+
+                        // BUild ON clause for join: ) AS CPx ON CPx.LOGICAL_RESOURCE_ID = CLR<x-1>.LOGICAL_RESOURCE_ID
+                        whereClauseSegment.append(RIGHT_PAREN + " AS " + chainedParmVar + ON)
+                                            .append(chainedParmVar + DOT + "LOGICAL_RESOURCE_ID" + EQ + prevChainedLogicalResourceVar + DOT + "LOGICAL_RESOURCE_ID");
+                    }
+                }
+            } else if (parmIndex == lastParmIndex) {
+                // This logic processes the LAST parameter in the chain.
+                if (!chainedParmProcessed) {
+                    if (!"_id".equals(currentParm.getCode())) {
+                        // Build this join: JOIN <modifierTypeResourceName>_<type>_VALUES AS CPx
+                        //                    ON CPx.LOGICAL_RESOURCE_ID = CLR<x-1>.LOGICAL_RESOURCE_ID
+                        //                    AND
+                        whereClauseSegment.append(JOIN + QuerySegmentAggregator.tableName(previousParm.getModifierResourceTypeName(), currentParm))
+                                            .append(" AS " + chainedParmVar + ON + chainedParmVar + DOT + "LOGICAL_RESOURCE_ID");
+                    } else {
+                        // Build this join: JOIN <modifierTypeResourceName>_LOGICAL_RESOURCES AS CLRx
+                        //                    ON CLRx.LOGICAL_RESOURCE_ID = CLR<x-1>.LOGICAL_RESOURCE_ID
+                        //                    AND
+                        whereClauseSegment.append(JOIN + previousParm.getModifierResourceTypeName() + "_LOGICAL_RESOURCES")
+                                            .append(" AS " + chainedLogicalResourceVar + ON + chainedLogicalResourceVar + DOT + "LOGICAL_RESOURCE_ID");
+                    }
+                    whereClauseSegment.append(EQ + prevChainedLogicalResourceVar + DOT + "LOGICAL_RESOURCE_ID" + AND);
+                }
+
+                // Build the rest
+                SqlQueryData sqlQueryData;
+                if (!"_id".equals(currentParm.getCode())) {
```

Comment:
You could embed the !chainedParmProcessed block down here, and only test this branch at most once

Net, I'm suggesting a logic change where the test `if (!"_id".equals(currentParm.getCode())) { ` is not in some circumstances twice, and branching of !chainedParmProcessed is done here before the existing code in this branch

P.S. I think this works, please validate.

Reply:
I think I did what you were suggesting here - please verify

Reply:
looks great - thanks Mike